### PR TITLE
refactor: Icon component

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This refactor allows us to more easily create snapshot tests for all Icons by doing the following:

- Store all valid icon names in an array
- Construct the `IconName` type based on that array
- Use that array to loop over the names and  dynamically create snapshot test for all valid icons.

### CODE REVIEW

**Comment by BouweCeunen** [comment]
Not 100% related to this PR, but maybe good to also make a story of the `chevron-left-square` icon in `Icon.stories.tsx`

**Comment by BartDeCaluwe** [comment]
> Not 100% related to this PR, but maybe good to also make a story of the `chevron-left-square` icon in `Icon.stories.tsx`

Yes definitely. I'm looking for a way to add those stories automatically, but that will be a different PR.

**Review by BouweCeunen — DISMISSED** [review]

**Comment by BouweCeunen** [comment]
> > Not 100% related to this PR, but maybe good to also make a story of the `chevron-left-square` icon in `Icon.stories.tsx`
> 
> Yes definitely. I'm looking for a way to add those stories automatically, but that will be a different PR.

Just as a follow-up, will we make a ticket for this or is it already part of one? Just so we don't forget.

**Comment by BartDeCaluwe** [comment]
> > > Not 100% related to this PR, but maybe good to also make a story of the `chevron-left-square` icon in `Icon.stories.tsx`
> > 
> > 
> > Yes definitely. I'm looking for a way to add those stories automatically, but that will be a different PR.
> 
> Just as a follow-up, will we make a ticket for this or is it already part of one? Just so we don't forget.

Made a ticket for it!

File: src/components/Icon.tsx [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
+	"bookmark",
+	"wallet",
+	"chevron-left-square",
+	"line-chart",
+	"coins",
+	"timeline",
+	"grid",
+	"user",
+] as const;
+export type IconName = typeof iconNames[number];
```
Comment (RuvendeGroote) [inline src/components/Icon.tsx:13]: cool, didn't know this

**Review by RuvendeGroote — APPROVED** [review]

**Review by BouweCeunen — APPROVED** [review]

**Pull request merged**